### PR TITLE
Add new condition on file uploads

### PR DESCRIPTION
**Pull request by gerrod3**
fixes: #729
Required PR: https://github.com/pulp/pulpcore/pull/2797

### CODE REVIEW

File: pulp_file/app/viewsets.py [hunk 1]
```diff
@@ -77,9 +77,15 @@ class FileContentViewSet(SingleArtifactContentUploadViewSet):
...
-                "action": ["list", "retrieve", "create"],
+                "action": ["list", "retrieve"],
+                "principal": "authenticated",
+                "effect": "allow",
+            },
+            {
+                "action": ["create"],
                 "principal": "authenticated",
                 "effect": "allow",
+                "condition": "has_required_repo_perms_on_upload:file.view_filerepository",
```
Comment (mdellweg) [inline pulp_file/app/viewsets.py:88]: View repository is IMHO not sufficient. This will create a repo version, so we need more Perm.

Comment (ipanova) [reply pulp_file/app/viewsets.py:88]: + `has_model_or_obj_perms:file.modify_filerepository`

File: pulp_file/tests/functional/utils.py [hunk 2]
```diff
@@ -121,6 +121,7 @@ def gen_artifact(url=FILE_URL, file=None):
         response = requests.get(url)
         with NamedTemporaryFile() as temp_file:
             temp_file.write(response.content)
+            temp_file.flush()
```
Comment (ipanova) [inline pulp_file/tests/functional/utils.py:124]: :+1:

File: pulp_file/app/viewsets.py (hunk 1 above)
Comment (gerrod3) [reply pulp_file/app/viewsets.py:88]: Just modify or do we need a new special upload permission?

File: pulp_file/app/viewsets.py [hunk 3]
```diff
@@ -77,9 +77,15 @@ class FileContentViewSet(SingleArtifactContentUploadViewSet):
...
-                "action": ["list", "retrieve", "create"],
+                "action": ["list", "retrieve"],
+                "principal": "authenticated",
+                "effect": "allow",
+            },
+            {
+                "action": ["create"],
                 "principal": "authenticated",
                 "effect": "allow",
+                "condition": "has_required_repo_perms_on_upload:file.modify_filerepository",
```
Comment (ipanova) [inline pulp_file/app/viewsets.py:88]: this is a nitpick, but if we follow the pattern we have established previously, once need to have modify *and* view repo perms. Same as - if you want to update a repo you need to have view and update perms

**Review by ipanova — APPROVED** [review]

**Pull request merged**